src/nvim/main.c

```diff
@@ -1439,6 +1445,28 @@ static void load_plugins(void)
 }
 
 /*
+ * :filetype plugin indent enable
+ */
+static void enable_filetype(mparm_T *parmp)
+{
+  if (parmp->use_vimrc != NULL && strcmp(parmp->use_vimrc, "NONE") == 0) {
+      return;
+  }
+  (void)force_enable_filetype();
+}
+
+/*
+ * Enable syntax highlighting.
+ */
+static void enable_syntax_highlighting(mparm_T *parmp)
+{
+  if (parmp->use_vimrc != NULL && strcmp(parmp->use_vimrc, "NONE") == 0) {
+      return;
+  }
+  (void)do_source((char_u *)SYNTAX_FILE, FALSE, DOSO_NONE);
+}
+
```

Comment:
Instead of all this, why not just do:
```c

// Don't do if `-u NONE` was passed on the command line
if (params->use_vimrc != NULL && strcmp(params->use_vimrc, "NONE") != 0) {
  // :filetype plugin indent enable
  (void)force_enable_filetype();
  // enable syntax highlighting
  (void)do_source((char_u *)SYNTAX_FILE, FALSE, DOSO_NONE);
}

```